frontend/src/deviceDetailsPage/configuration/UploadConfigDialog.jsx

```diff
@@ -30,13 +30,14 @@ export default class UploadConfigDialog extends React.Component {
   handleUpload() {
     const { deviceId } = this.props;
     const { file } = this.state;
-    uploadConfiguration(deviceId, file).then(() => {
-      this.afterUpload();
-    });
-  }
-
-  afterUpload() {
-    this.dialogElement.current.closeDialog();
+    uploadConfiguration(deviceId, file)
+      .then(() => {
+        this.dialogElement.current.closeDialog();
```

Comment:
Shouldn't the dialog always be closed after clicking? It feels a bit weird to have the dialog still hanging around after clicking it even if an error snackbar appears afterwards.

Reply:
good catch. Moved close Dialog to finally block in fa40b11